File: api/src/migrations/libsql/005_utm.ts
import { getLibsqlRepo } from "../../repository/repo";

async function migrate(): Promise<boolean> {
  const db = getLibsqlRepo().db();

  await db.executeMultiple(`
    ALTER TABLE events ADD COLUMN utm_source TEXT;
    ALTER TABLE events ADD COLUMN utm_medium TEXT;
    ALTER TABLE events ADD COLUMN utm_campaign TEXT;
    ALTER TABLE events ADD COLUMN utm_term TEXT;
    ALTER TABLE events ADD COLUMN utm_content TEXT;
  `);

  return true;
}

export default migrate;

File: backend/migrations/005_utm.go
package migrations

import (
	"joinstorywise.com/api/db"
)

var AddUTMColumnsMigration = Migration{
	ID: "005_utm",
	Execute: func(repo *db.PostgresRepo) error {
		_, err := repo.Db.Exec(`
			ALTER TABLE events
			ADD COLUMN IF NOT EXISTS utm_source TEXT,
			ADD COLUMN IF NOT EXISTS utm_medium TEXT,
			ADD COLUMN IF NOT EXISTS utm_campaign TEXT,
			ADD COLUMN IF NOT EXISTS utm_term TEXT,
			ADD COLUMN IF NOT EXISTS utm_content TEXT;
		`)
		if err != nil {
			return err
		}

		return nil
	},
}
